memo/managers.py

```diff
@@ -73,3 +74,63 @@ def get_card_amount_in_deck(self, deck):
 class DeckManager(models.Manager):
     def get_decks_by_owner(self, user):
         return self.filter(owner=user)
+
+
+class CardAnswerHistoryManager(models.Manager):
+    def get_consecutive_days(self, user):
+        from .models import Card
+
+        max_daily_date_times = (
+            self.filter(owner=user)
+            .extra(select={"day": "date( created_at )"})
+            .values("day")
+            .annotate(latest_datetime=Max("created_at"))
+        )
```

Comment:
When making a query with django, we don't need to call `.values` in order to get that value. the [values](https://docs.djangoproject.com/en/4.0/ref/models/querysets/#django.db.models.query.QuerySet.values) method is going to return a list of dicts instead of a list of objects. Ex:

```python
# This list contains a Blog object.
>>> Blog.objects.filter(name__startswith='Beatles')
<QuerySet [<Blog: Beatles Blog>]>

# This list contains a dictionary.
>>> Blog.objects.filter(name__startswith='Beatles').values()
<QuerySet [{'id': 1, 'name': 'Beatles Blog', 'tagline': 'All the latest Beatles news.'}]>
```
With a list of objects, you can access the attributes of the table like this `Blog.title` instead of `Blog.get("title")`. This is good to avoid misspelling and you also type less =].
So my suggestion is: remove the `values` method, remove the `extra` since the `day` value is not been used (by the way, nice work finding this feature, I actually had to look it up to see what it does) and replace the `annotate` for `order_by` since it's going to do the same thing.
```suggestion
        max_daily_date_times = self.filter(owner=user).order_by("created_at")
```

Reply:
Hello, @r-lelis! Thank you for your comment and for giving an example about queryset. 

I'm using `.extra(select={"day": "date( created_at )"})` to get the date without the time because if I group only by DateTime (created_at without `date() or cast()`), it'll try to group only by date and time and that's not exactly what I need there. For example, if I run the code as the suggested change, it will return like: 

| CARD_ID      | CREATED_AT |
|:--------|:---------|
| 4  | 2022-08-04 04:41:59.605 -0300  |
| 3 | 2022-08-02 05:30:59.605 -0300  |
| 2   | 2022-08-01 12:41:59.605 -0300   |
| 1   | 2022-08-01 16:32:59.605 -0300   |

(And all the other columns)

I can't repeat the date because it will iterate in line 109 `for index in range(0, len(objects)):`.

So, I need it to return like this:


| DAY      | CREATED_AT |
|:--------|:---------|
| 2022-08-04 | 2022-08-04 04:41:59.605 -0300  |
| 2022-08-02 | 2022-08-02 05:30:59.605 -0300  |
| 2022-08-01 | 2022-08-01 16:32:59.605 -0300   |

It grouped by date and returned the latest datetime of that day. 

This is what I'm trying to do in the `max_daily_date_times`:
```
SELECT date(created_at) AS day,
       MAX(created_at) AS latest_datetime
FROM memo_cardanswerhistory 
WHERE owner_id = {user_id}
GROUP BY date( created_at )
```

Reply:
Oooh, you were grouping by the day! Now I see, really nice. All right then, leave it as it is =]